services/galley/src/Galley/API/Query.hs

```diff
@@ -148,11 +151,65 @@ getUnqualifiedConversation ::
   Local UserId ->
   ConvId ->
   Sem r Public.ConversationV9
-getUnqualifiedConversation lusr cnv = do
+getUnqualifiedConversationV9 lusr cnv = do
   c <- getConversationAndCheckMembership (tUntagged lusr) (qualifyAs lusr cnv)
   Mapping.conversationViewV9 lusr c
 
+getUnqualifiedConversation ::
+  forall r.
+  ( Member ConversationStore r,
+    Member (ErrorS 'ConvNotFound) r,
+    Member (ErrorS 'ConvAccessDenied) r,
+    Member TeamStore r
+  ) =>
+  Local UserId ->
+  ConvId ->
+  Sem r Public.Conversation
+getUnqualifiedConversation lusr cnv = do
+  mChannel <- getChannel lusr cnv
+  case mChannel of
+    Just channel -> pure channel
+    Nothing ->
+      Mapping.conversationView (qualifyAs lusr ()) (Just lusr)
+        <$> getConversationAndCheckMembership (tUntagged lusr) (qualifyAs lusr cnv)
```

Comment:
Why do we get the conv data twice here in case it is not a channel? Perhaps we can change the `getConversationAndCheckMembership` to allow an admin to succeed the membership check?